serverstatus/serverstatus.go

```diff
@@ -132,24 +150,43 @@ func sendEmbeddedMessage(roomID string, color int, title, description string) {
 	bot.Session.ChannelMessageSendEmbed(roomID, embed)
 }
 
-// MessageHandler will be called every time a new
-// message is created on any channel that the autenticated bot has access to.
-func MessageHandler(s *discordgo.Session, m *discordgo.MessageCreate) {
-
-	// Ignore all messages created by the bot itself
-	if m.Author.ID == bot.BotID {
-		return
-	}
-
-	if m.Content == config.Config.BotPrefix+"ServerStatus" {
-		for _, server := range config.Config.Servers {
-			if server.Online {
-				sendEmbeddedMessage(m.ChannelID, green, server.Name, "Online!\nUptime: "+fmtDuration(time.Since(server.OnlineTimestamp)))
-			} else {
-				sendEmbeddedMessage(m.ChannelID, red, server.Name, "Offline!\nDowntime: "+fmtDuration(time.Since(server.OfflineTimestamp)))
-			}
-		}
-	}
+// InteractionHandler will be called every time an interaction from a user occurs
+// Command interaction handling requires bot command scope
+func InteractionHandler(s *discordgo.Session, i *discordgo.InteractionCreate) {
+    // A user is calling us with our status command
+    if i.ApplicationCommandData().Name == "server-status" {
+        online := ""
+        offline := ""
+
+        for _, server := range config.Config.Servers {
+            if server.Online {
+                online = online + server.Name + "  :  " + fmtDuration(time.Since(server.OnlineTimestamp)) + "\n"
+            } else {
+                offline = offline + server.Name + "  :  " + fmtDuration(time.Since(server.OfflineTimestamp)) + "\n"
+            }
+        }
+
+        // Only one message can be an interaction response. Messages can only contain up to 10 embeds.
+        // Our message will therefore instead be two embeds (online and offline), each with a list of servers in text.
+        // Embed descriptions can be ~4096 characters, so no limits should get hit with this.
+        s.InteractionRespond(i.Interaction, &discordgo.InteractionResponse {
+            Type: discordgo.InteractionResponseChannelMessageWithSource,
+            Data: &discordgo.InteractionResponseData {
+                Embeds: []*discordgo.MessageEmbed {
+                    {
+                        Title: ":white_check_mark: Online",
+                        Color: green,
+                        Description: online,
+                    },
+                    {
+                        Title: ":x: Offline",
```

Comment:
Looks like this will still send an offline message if all servers are online and vice versa with online servers. I can take care of this in a future commit.

Reply:
It seems to work as expected in my tests, but admittedly this was written pretty quickly so improvements will be great
![2024-03-03_09-40](https://github.com/mgerb/ServerStatus/assets/11908638/d6feeb16-8d51-4757-a8b7-8401e97c0835)

Reply:
If your offline server list was empty it was still sending the offline message with no servers in it. Looks like you still have some offline servers in this list. No worries though - I fixed it here https://github.com/mgerb/ServerStatus/commit/50a06b98be6ac706184a20f326910fd6419d553f#diff-2ca645d10ad115ebfe751c0e2bf39c5fbd65367fc33b6f5d7fee3fd29bd62fddR179

Reply:
> If your offline server list was empty it was still sending the offline message with no servers in it. Looks like you still have some offline servers in this list. No worries though - I fixed it here [50a06b9#diff-2ca645d10ad115ebfe751c0e2bf39c5fbd65367fc33b6f5d7fee3fd29bd62fddR179](https://github.com/mgerb/ServerStatus/commit/50a06b98be6ac706184a20f326910fd6419d553f#diff-2ca645d10ad115ebfe751c0e2bf39c5fbd65367fc33b6f5d7fee3fd29bd62fddR179)

Oh I understand now, you're correct. I figured it looked ok to visually show the user even if there's nothing that is offline / online, but excluding empty embeds looks like a good decision.